packages/bsp/common/etc/profile.d/armbian-check-first-login.sh

```diff
@@ -16,29 +16,40 @@ check_abort()
 	exit 0
 }
 
-set_timezone()
+
+set_timezone_and_locales()
 {
 	# Grab this machine's public IP address
 	PUBLIC_IP=`curl -s https://ipinfo.io/ip`
 	if [ $? -eq 0 ]; then
 		while true; do
-		echo ""
-		read -p "\nDo you wish to determine time zone from your location? [Yes]" yn
+		echo "\n"
+		read -p "Do you wish to determine time zone from your location? [Yes]" yn
 		if [ -z $yn ]; then yn=yes; fi
 		case $yn in
 	        [Yy]* )
-		echo -e "\nSetting Timezone based on your geo location with help from ipinfo.io"
+		echo -e "\nSetting Timezone based on your geo location with help from http://ipwhois.app"
 		# Call the geolocation API and capture the output
-		TZ=$(curl -s https://ipvigilante.com/${PUBLIC_IP} | \
-	        jq '.data.continent_name, .data.city_name' | \
-
-		while read -r CONTINENT; do
-	                read -r CITY
-	                echo "${CONTINENT}/${CITY}" | \
-	                        tr --delete \"
-		done)
-		timedatectl set-timezone "${TZ}"
-		dpkg-reconfigure --frontend=noninteractive tzdata
+		RES=$(
+			curl -s http://ipwhois.app/json/${PUBLIC_IP} | \
```

Comment:
Maybe it is a good idea to add a timeout here?

Reply:
Absolutely.